### PR TITLE
actually use number to keep track of resources and resource attributes

### PR DESCRIPTION
**Pull request by motivator**

### CODE REVIEW

File: src/clusto/drivers/base/resourcemanager.py [hunk 1]
```diff
@@ -195,12 +200,20 @@ def deallocate(self, thing, resource=(), number=True):
...
+                    res_mgr = self.get_resource_manager(res)
+                    if res_mgr != self:
+                        continue
                     thing.del_attrs(self._attr_name, number=res.number)
 
             elif resource and not self.available(resource, number):
                 resource, number = self.ensure_type(resource, number)
+                try:
+                    number = self.get_resource_number(thing, resource)
+                except:
```
Comment (plathrop) [inline src/clusto/drivers/base/resourcemanager.py]: Blanket `except` clauses are generally dangerous in python. You should probably enumerate the exceptions you are actually concerned about instead.

**Comment by plathrop** [comment]
minor nit, otherwise looks good

File: src/clusto/drivers/base/resourcemanager.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 from clusto.schema import select, and_, ATTR_TABLE, Attribute, func, Counter
 from clusto.drivers.base import Driver, ClustoMeta
 from clusto.exceptions import ResourceTypeException, ResourceNotAvailableException, ResourceException
+import logging
```
Comment (thekad) [inline src/clusto/drivers/base/resourcemanager.py]: remove this import

**Comment by thekad** [comment]
:+1:

**Pull request merged**